src/Map.php

```diff
@@ -163,4 +163,19 @@ public function parse(array $map) {
         $this->logger->info('Swivel - Parsing feature map.', compact('map'));
         return array_combine(array_keys($map), array_map([$this, 'reduceToBitmask'], $map));
     }
+
+    public static function __set_state(array $stateArray)
+    {
+        if (!array_key_exists('map', $stateArray)) {
+            throw new \RuntimeException('"map" element does not exist in state array');
+        }
+
+        $map = new static();
```

Comment:
@mrferos Since you're constructing here, is there a reason not to pass the `$stateArray['map']` to the constructor instead of using a reflection to expose the protected property?